lib/monitor.toit

```diff
@@ -222,12 +221,23 @@ monitor Channel:
   The order in which waiting tasks are unblocked is unspecified.
   */
   receive --blocking/bool=true -> any:
-    if not blocking and c_ == p_: return null
-    await: c_ != p_
-    value := buffer_[c_]
-    c_ = (c_ + 1) % buffer_.size
+    if not blocking and buffered_count_ == 0: return null
+    await: buffered_count_ > 0
+    value := buffer_[start_]
+    start_ = (start_ + 1) % buffer_.size
+    buffered_count_--
     return value
 
+  /**
+  The capacity of the channel.
+  */
+  capacity -> int: return buffer_.size
+
+  /**
+  The amount of messages that are currently queued in the channel.
+  */
+  buffered_count -> int: return buffered_count_
```

Comment:
I kinda like just calling this size. What do you think?

Reply:
I'm hesitant. I can easily see myself asking for "a channel of size 8".
To me the "size" often refers to the capacity of the channel, and not how many items are in it.

Reply:
Changed to `size`.
done.